src/ansys/dpf/core/data_sources.py

```diff
@@ -116,15 +116,26 @@ def set_result_file_path(self, filepath, key=""):
         ['/tmp/file.rst']
 
         """
-        # Handle the case of files without extension, such as the LS-DYNA d3plot file
-        if os.path.splitext(filepath)[1] == "":
-            if "d3plot" in os.path.basename(filepath):
-                key = "d3plot"
+        # Handle no key given and no file extension
+        if key == "" and os.path.splitext(filepath)[1] == "":
+            key = self.guess_result_key(str(filepath))
         if key == "":
             self._api.data_sources_set_result_file_path_utf8(self, str(filepath))
         else:
             self._api.data_sources_set_result_file_path_with_key_utf8(self, str(filepath), key)
 
+    @staticmethod
+    def guess_result_key(filepath: str) -> str:
+        """Guess result key for files without a file extension."""
+        d3plot_key = "d3plot"
+        binout_key = "binout"
+        # Handle files without extension
+        if d3plot_key in os.path.basename(filepath):
+            return d3plot_key
+        elif binout_key in os.path.basename(filepath):
+            return binout_key
```

Comment:
Maybe we could generalize this in a list ["d3plot", "binout"], and loop over it, so it can be easily extensible